plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -3491,4 +3503,331 @@ public void testQueriesWithoutCheckpointFiltering()
+        testTimestampCoercionOnCreateTableAsWithMapTypeFailure("TIMESTAMP '1970-01-01 00:00:00.111222333444'", ".*Unsupported type: timestamp\\(12\\).*");
+        testTimestampCoercionOnCreateTableAsWithMapTypeFailure("TIMESTAMP '1970-01-01 00:00:00.9999995'", ".*Unsupported type: timestamp\\(7\\).*");
+        testTimestampCoercionOnCreateTableAsWithMapTypeFailure("TIMESTAMP '1970-01-01 23:59:59.9999995'", ".*Unsupported type: timestamp\\(7\\).*");
+        testTimestampCoercionOnCreateTableAsWithMapTypeFailure("TIMESTAMP '1969-12-31 23:59:59.9999995'", ".*Unsupported type: timestamp\\(7\\).*");
+        testTimestampCoercionOnCreateTableAsWithMapTypeFailure("TIMESTAMP '1969-12-31 23:59:59.999999499999'", ".*Unsupported type: timestamp\\(12\\).*");
+        testTimestampCoercionOnCreateTableAsWithMapTypeFailure("TIMESTAMP '1969-12-31 23:59:59.9999994'", ".*Unsupported type: timestamp\\(7\\).*");
+    }
+
+    private void testTimestampCoercionOnCreateTableAsWithMapType(String actualValue, String expectedValue)
+            throws Exception
+    {
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_timestamp_coercion_on_create_table_as_with_map_type",
+                "AS SELECT map(array['key'], array[%s]) ts".formatted(actualValue))) {
+            assertThat(getColumnType(testTable.getName(), "ts")).isEqualTo("map(varchar, timestamp(6))");
+            assertThat(query("SELECT ts['key'] FROM " + testTable.getName()))
+                    .skippingTypesCheck()
+                    .matches("VALUES " + expectedValue);
+            assertTimestampNtzFeature(testTable.getName());
+        }
+    }
+
+    private void testTimestampCoercionOnCreateTableAsWithMapTypeFailure(String actualValue, @Language("RegExp") String expectedMessage)
+    {
+        assertQueryFails(
+                "CREATE TABLE test_timestamp_coercion_on_create_table_as_with_map_type AS SELECT map(array['key'], array[%s]) ts".formatted(actualValue),
+                expectedMessage);
+    }
+
+    private void assertTimestampNtzFeature(String tableName)
+            throws Exception
+    {
+        // Verify reader/writer version and features in ProtocolEntry
+        String tableLocation = getTableLocation(tableName);
+        List<DeltaLakeTransactionLogEntry> transactionLogs = getEntriesFromJson(0, tableLocation + "/" + TRANSACTION_LOG_DIRECTORY, FILE_SYSTEM).orElseThrow();
+        ProtocolEntry protocolEntry = transactionLogs.get(1).getProtocol();
+        assertThat(protocolEntry).isNotNull();
+        assertThat(protocolEntry.getMinReaderVersion()).isEqualTo(3);
+        assertThat(protocolEntry.getMinWriterVersion()).isEqualTo(7);
+        assertThat(protocolEntry.getReaderFeatures()).isEqualTo(Optional.of(ImmutableSet.of("timestampNtz")));
+        assertThat(protocolEntry.getWriterFeatures()).isEqualTo(Optional.of(ImmutableSet.of("timestampNtz")));
```

Comment:
I was missing we can verify these information with `$properties` system table. Sorry for my previous suggestion.